core/backend/src/ElementTreeWalker.ts

```diff
@@ -18,6 +17,23 @@ const loggerCategory = `${BackendLoggerCategory.IModelDb}.ElementTreeWalker`;
 /** @beta */
 export interface ElementTreeWalkerModelInfo { model: Model, isDefinitionModel: boolean }
 
+function sortChildrenBeforeParents(iModel: IModelDb, ids: Id64Array): Array<Id64Array> {
+  const children: Id64Array = [];
+  const parents: Id64Array = [];
+  for (const eid of ids) {
+    const el = iModel.elements.getElement(eid);
```

Comment:
Is it necessary to load the entire (potentially very large) element when all you want is the parent Id, vs querying directly for parent Id?

Reply:
Good point. I added IModelDb.Elements.queryParent